vernac/classes.ml

```diff
@@ -27,12 +27,23 @@ open Libobject
 module RelDecl = Context.Rel.Declaration
 module NamedDecl = Context.Named.Declaration
 (*i*)
+let warn_default_modes = CWarnings.create ~name:"class-declaration-default-mode" ~category:"automation"
+  ~default:CWarnings.Disabled
+  Pp.(fun (gr, m) -> hov 2 (str "Using inferred default " ++
+    str (CString.plural (List.length m) "mode") ++ str":" ++ spc () ++
+    prlist_with_sep pr_comma (prlist_with_sep spc Hints.pp_hint_mode) m ++
+    spc () ++ str "for" ++ spc () ++ Printer.pr_global gr))
 
 let set_typeclass_transparency c local b =
   let locality = if local then Goptions.OptLocal else Goptions.OptGlobal in
   Hints.add_hints ~locality [typeclasses_db]
     (Hints.HintsTransparencyEntry (Hints.HintsReferences [c], b))
 
+let set_typeclass_modes c local modes =
+  let locality = if local then Goptions.OptLocal else Goptions.OptGlobal in
+  List.iter (fun mode -> Hints.add_hints ~locality [typeclasses_db]
+    (Hints.HintsModeEntry (c,mode))) modes
```

Comment:
This function is actually only used with `local` set to `false`, so its code could (and probably should) be simplified. Furthermore, I'm concerned that we add *global* hints when *export* hints are now supported and should usually be preferred (and will eventually become the new default when the `Hint Mode` command is used). The doc should also be clarified on that aspect.